### ISSUE TITLE
error: 'CGDisplayCreateImageForRect' is unavailable: obsoleted in macOS 15.0 - Please use ScreenCaptureKit instead

### ISSUE THREAD

**Issue by ryandesign** [aseprite/laf#108]
Building laf, as part of building Aseprite 1.3.8.1, fails when using the macOS 15.0 deployment target, which is the default when building on macOS 15:

```
laf/os/osx/system.mm:118:22: error: 'CGDisplayCreateImageForRect' is unavailable: obsoleted in macOS 15.0 - Please use ScreenCaptureKit instead.
  118 |   CGImageRef image = CGDisplayCreateImageForRect(CGMainDisplayID(),
      |                      ^
```

A workaround is to tell the build to use the macOS 14.0 deployment target (or earlier) by adding the flag `-DCMAKE_OSX_DEPLOYMENT_TARGET=14.0` when running `cmake`.

**Comment by dacap** [comment]
Hi @ryandesign, in our instructions (at least on Aseprite INSTALL guide) we use:
```
      -DCMAKE_OSX_ARCHITECTURES=x86_64 \
      -DCMAKE_OSX_DEPLOYMENT_TARGET=10.9 \
```
when running cmake, probably if you are on Apple Silicon you will want to use 11.0:
```
      -DCMAKE_OSX_ARCHITECTURES=arm64 \
      -DCMAKE_OSX_DEPLOYMENT_TARGET=11.0 \
```
we should always use the oldest platform possible on macOS.

**Comment by ryandesign** [comment]
Yes, if you're distributing a binary to others, you want it to have the widest possible compatibility by using the oldest possible deployment target.

Users building for their own use on their own system don't care about that.

Using an older deployment target does work around the error but you will at some point want to update your code to build without errors when using the macOS 15 or later SDK.
